edb/server/protocol/protocol.pyx

```diff
@@ -189,13 +231,89 @@ cdef class HttpProtocol:
             response.body,
             response.close_connection)
 
+    def _switch_to_binary_protocol(self, data=None):
+        binproto = binary.EdgeConnection(self.server, self.external_auth)
+        self.transport.set_protocol(binproto)
+        binproto.connection_made(self.transport)
+        if data:
+            binproto.data_received(data)
+        return binproto
+
+    def _init_http_parser(self):
+        self.parser = httptools.HttpRequestParser(self)
+        self.current_request = HttpRequest()
+
+    async def _forward_first_data(self, data):
+        # As we stole the "first data", we need to manually send it back to
+        # the SSLProtocol. The hack here is highly-coupled with uvloop impl.
```

Comment:
Perhaps we should add an explicit method for this in uvloop, like `inject_data` or something. But that can be done separately if you think it's a good idea.